pkg/cloudproviders/aws/metadata.go

```diff
@@ -114,3 +123,56 @@ func plaintextIdentityDoc(ctx context.Context, mdClient *ec2metadata.EC2Metadata
 
 	return doc, nil
 }
+
+// getClusterMetadata attempts to get the EKS cluster name on a best effort basis.
+// First, it tries to get the cluster name from the EC2 instance tags. Access to
+// the tags must be explicitly enabled for the EC2 instance beforehand.
+// Second, it tries the node labels. The label is only set when the EKS cluster
+// was created via eksctl.
+func getClusterMetadata(ctx context.Context,
+	client *ec2metadata.EC2Metadata, doc *ec2metadata.EC2InstanceIdentityDocument,
+) *storage.ClusterMetadata {
+	clusterName, err := getClusterNameFromInstanceTags(ctx, client)
+	if err == nil {
+		return clusterMetadataFromName(clusterName, doc)
+	}
+	log.Errorf("Failed to get EKS cluster metadata from instance tags: %v", err)
```

Comment:
Hmm, how frequently is this done / executed? If it's a one-off thing, it's OK, in case this is frequently attempted, might be best to use the rate limited logger to avoid too many occurrences of this.

Reply:
The metadata is part of the telemetry gatherer, probably better to add a rate limiter.